en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/extensions/custom-filters.md

```diff
@@ -0,0 +1,113 @@
+the enforcer starts, the JAR files in that directory are added to the classpath. Using Java Service Provider
+Interface, the classloading happens in the enforcer. See the following sections for further details on how to implement this.
+ 
+## How to add a custom filter
+
+1. Create a Java project with `org.wso2.choreo.connect.enforcer.commons` dependency.
+   - For Apache Maven, use the following.
+   ```xml
+   <dependency>
+       <groupId>org.wso2.am.choreo.connect</groupId>
+       <artifactId>org.wso2.choreo.connect.enforcer.commons</artifactId>
+       <version>1.0.0</version>
+   </dependency>
+   ```
+
+2. Use the following interface to implement the custom filter.
+
+    ```java
+    package org.wso2.choreo.connect.enforcer.commons;
+    
+    import org.wso2.choreo.connect.enforcer.commons.model.APIConfig;
+    import org.wso2.choreo.connect.enforcer.commons.model.RequestContext;
+    
+    import java.util.Map;
+    
+    /**
+        * Filters are the request interceptors that applies API Management capabilities at the gateway layer. This is the
+        * Interface to implement chain of filters at the gateway.
+        */
+    public interface Filter {
+    
+        default void init(APIConfig apiConfig, Map<String, String> configProperties){};
+    
+        boolean handleRequest(RequestContext requestContext);
+    }
+    ```
+
+    Here is the sample filter implementation that reads the property called `CustomProperty` from Filter 
+    Configuration and adds it as a header to the request.
+
+    ```java
+    package org.example.tests;
+    
+    import org.wso2.choreo.connect.enforcer.commons.model.APIConfig;
+    import org.wso2.choreo.connect.enforcer.commons.model.RequestContext;
+    import org.wso2.choreo.connect.enforcer.commons.Filter;
+    
+    import java.util.Map;
+    
+    public class CustomFilter implements Filter {
+        private static final Logger log = LoggerFactory.getLogger(CustomFilter.class);
+        private Map<String, String> configProperties;
+    
+        @Override
+        public void init(APIConfig apiConfig, Map<String, String> configProperties) {
+            this.configProperties = configProperties;
+        }
+    
+        @Override
+        public boolean handleRequest(RequestContext requestContext) {
+            String headerValue = configProperties.get("CustomProperty");
+            requestContext.addOrModifyHeaders("Custom-header-1", headerValue);
+            return true;
+        }
+    }
+    ```
+
+3. Since we use Java SPI (Service Provider Interface), we need to provide the provider configuration file
+`META-INF/services/org.wso2.choreo.connect.enforcer.commons.Filter`. If you are using Apache Maven, create the
+file inside the `<PROJECT>/src/main/resources` directory. The content of the file needs to be the qualified class name
+of the filter implementation.
+
+    ```
+    org.example.tests.CustomFilter
+    ```
+
+4. Build the project and create the JAR file.
```

Comment:
Let's add instructions on how to do this or point them to a page that has this information.